### PR TITLE
CursorSync: Extract EventBus logic into shared plugin

### PR DESCRIPTION
**Pull request by leeoniya**
this simplifies/DRYs the code, makes removal of old tooltips code easier (https://github.com/grafana/grafana/issues/83632), and improves performance by throttling events before publish instead of after subscribe.

the new `syncAny` as well as existing `syncTooltip` callbacks will be removed soon in favor of simply re-initializing the viz when shared tooltip/crosshair option is changed in the dashboard settings. this PR just makes things uniform in all panels for the time being.

tested with and without `newVizTooltips` plus `tableSharedCrosshair`.

enable `newVizTooltips` and `tableSharedCrosshair`, then import this:

test dashboard: [eventbus-test.json](https://github.com/grafana/grafana/files/14532960/eventbus-test.json)

### CODE REVIEW

File: packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx [hunk 1]
```diff
@@ -501,9 +501,9 @@ export const TooltipPlugin2 = ({
       }
     });
 
-    const onscroll = () => {
+    const onscroll = (e: Event) => {
       updatePlotVisible();
-      _isHovering && !_isPinned && dismiss();
+      _isHovering && !_isPinned && e.target?.contains(_plot!.root) && dismiss();
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx:506]: this improves tooltip hiding behavior by only hiding when an ancestor is scrolled, and not layout-unrelated panels.

File: public/app/plugins/panel/candlestick/CandlestickPanel.tsx [hunk 2]
```diff
@@ -260,19 +282,11 @@ export const CandlestickPanel = ({
...
-      {(uplotConfig, alignedDataFrame) => {
-        alignedDataFrame.fields.forEach((field) => {
-          field.getLinks = getLinksSupplier(
-            alignedDataFrame,
-            field,
-            field.state!.scopedVars!,
-            replaceVariables,
-            timeZone
-          );
-        });
```
Comment (leeoniya) [inline public/app/plugins/panel/candlestick/CandlestickPanel.tsx:272]: unrelated cleanup. (forgot to remove in https://github.com/grafana/grafana/pull/83654)

File: public/app/plugins/panel/candlestick/CandlestickPanel.tsx [hunk 3]
```diff
@@ -260,19 +282,11 @@ export const CandlestickPanel = ({
...
-
+      {(uplotConfig, alignedFrame) => {
         return (
           <>
+            <KeyboardPlugin config={uplotConfig} />
```
Comment (leeoniya) [inline public/app/plugins/panel/candlestick/CandlestickPanel.tsx:288]: same as TimeSeries

**Comment by dprokop** [comment]
Looks good, think getting rid of the sync config from GraphNG is a wise move.

File: packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts [hunk 4]
```diff
@@ -306,7 +297,6 @@ type UPlotConfigPrepOpts<T extends Record<string, unknown> = {}> = {
   theme: GrafanaTheme2;
   timeZones: TimeZone[];
   getTimeRange: () => TimeRange;
-  eventBus: EventBus;
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts:309]: RIP

**Review by adela-almasan — APPROVED** [review]
LGTM
